### PR TITLE
Nldas grid

### PR DESCRIPTION
**Pull request by billsacks**
### Description of changes

Add support for NLDAS grid

### Specific notes

Contributors other than yourself, if any: @barlage helped with some file creation

CTSM Issues Fixed (include github issue #): none

Are answers expected to change (and if so in what way)? no

Any User Interface Changes (namelist or namelist defaults changes)? none

Testing performed, if any:
SMS_D.nldas2_mnldas2.I2000Clm50SpGs.cheyenne_intel.clm-default

### CODE REVIEW

**Comment by billsacks** [comment]
@ekluzek - I plan to bring this to master combined with the NWP configuration (i.e., combining them into a single branch and making a tag from that). And before I bring this to master, I'll make tags of cime and mosart. So this isn't quite in its final form. But I think all of the necessary CTSM changes for the NLDAS grid are now in place, so I wanted to give you a chance to give this a quick look, if you want.

File: bld/namelist_files/namelist_defaults_overall.xml [hunk 1]
```diff
@@ -75,6 +75,7 @@ determine default values for namelists.
 <mask hgrid="ne240"               >gx1v6</mask>
 <mask hgrid="ne30"                >gx1v6</mask>
 
+<mask hgrid="0.125nldas2"         >nldas2</mask>
```
Comment (billsacks) [inline bld/namelist_files/namelist_defaults_overall.xml]: I couldn't tell if this was actually needed, but it seems to be done for other grids, so I added it.

File: python/README.md [hunk 2]
```diff
@@ -20,4 +20,7 @@ support different options:
    
 ## pylint
 
-You can run pylint on everything in the ctsm package with `make lint`
+You can run pylint on everything in the ctsm package with `make lint`.
```
Comment (billsacks) [inline python/README.md]: Unrelated change

File: tools/mksurfdata_map/Makefile.data [hunk 3]
```diff
@@ -128,6 +129,12 @@ global-present-ne16np4 : FORCE
 global-present-ne120np4 : FORCE
 	$(MKSURFDATA) -no-crop -glc_nec 10 -y 2000 -res ne120np4 $(BACKGROUND)
 
+# BUG(wjs, 2019-04-10, ESCOMP/ctsm#262) This -hirespft won't work until issue 262 is resolved
+# For now, I'm using:
+#    -no-crop -glc_nec 10 -y 2005 -res 0.125nldas2
+global-present-nldas : FORCE
+	$(MKSURFDATA) -no-crop -hirespft -glc_nec 10 -y 2005 -res 0.125nldas2 $(BACKGROUND)
```
Comment (billsacks) [inline tools/mksurfdata_map/Makefile.data]: I debated whether to write this rule according to what I used for the current dataset or according to what we eventually want. I decided on the latter, with the assumption that #262 will be fixed relatively soon, because I was concerned that, if I wrote it the other way, we'd forget to change it when it should be changed. But I'm open to changing this if you'd like.

**Comment by billsacks** [comment]
cime changes are documented in https://github.com/ESMCI/cime/pull/3063 (Note that that uses a rebased version of the cime branch, but the changes are the same as the ones included here.)

**Comment by billsacks** [comment]
@ekluzek per your request, I have run mkmapdata.sh with a global and regional grid:

For global:

```bash
cmdargs="-r 10x15 -f /glade/p/cesmdata/cseg/inputdata/lnd/clm2/mappingdata/grids/SCRIPgrid_10x15_nomask_c110308.nc -t global"
regrid_num_proc=8
echo "args: $cmdargs"
echo "time env REGRID_PROC=$regrid_num_proc ./mkmapdata.sh $cmdargs\n"
time env REGRID_PROC=$regrid_num_proc ./mkmapdata_test.sh $cmdargs
```

For single point:

```bash
cmdargs="-r 1x1_brazil -f /glade/p/cesmdata/cseg/inputdata/lnd/clm2/mappingdata/grids/SCRIPgrid_1x1pt_brazil_nomask_c110308.nc -t regional"
regrid_num_proc=1
echo "args: $cmdargs"
echo "time env REGRID_PROC=$regrid_num_proc ./mkmapdata.sh $cmdargs\n"
time env REGRID_PROC=$regrid_num_proc ./mkmapdata_test.sh $cmdargs
```

In both cases, I changed mkmapdata's grid list to just contain one grid:

```bash
    grids=(                    \
           "0.5x0.5_AVHRR"     \
          )
```

File: bld/namelist_files/namelist_defaults_overall.xml (hunk 1 above)
Comment (ekluzek) [reply bld/namelist_files/namelist_defaults_overall.xml]: It is right now. It listens to this, but needs to be changed to listen to the actual grid.

File: tools/mksurfdata_map/Makefile.data (hunk 3 above)
Comment (ekluzek) [reply tools/mksurfdata_map/Makefile.data]: I agree with this one.

**Comment by billsacks** [comment]
> I think this all looks good. Before coming to master, I'd like to see the external hashes changed to tags, and the input files all checked into inputdata (I checked the SCRIPgrid file and it wasn't, but didn't check others).

Yup, working on both of those things now. (I wanted to wait for review approval before adding the files to the inputdata repo.)
